### PR TITLE
Lambda functions development

### PR DESCRIPTION
**Pull request by mregger**

### CODE REVIEW

File: lambda-functions/iot-rds-integration-cnf.json [hunk 1]
```diff
@@ -0,0 +1,208 @@
+
```
Comment (rjarsenault1101) [inline lambda-functions/iot-rds-integration-cnf.json]: Delete JSON file

File: lambda-functions/iot-rds-integration.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+def make_connection():
+    return pymysql.connect(host=endpoint, user=dbuser, passwd=password,
+        port=int(port), db=database, autocommit=True, cursorclass=pymysql.cursors.DictCursor)
+
+def log_err(errmsg):
+    logger.error(errmsg)
+    return {"body": errmsg , "headers": {}, "statusCode": 400,
+        "isBase64Encoded":"false"}
+
+logger.info("Cold start complete.")
```
Comment (rjarsenault1101) [inline lambda-functions/iot-rds-integration.py]: delete

File: lambda-functions/iot-rds-integration.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+def handler(event,context):
+
+    #logger.info('Data received: ', str(event))
```
Comment (rjarsenault1101) [inline lambda-functions/iot-rds-integration.py]: delete commented out code

File: lambda-functions/rds-api-cnf.json [hunk 4]
```diff
@@ -0,0 +1,208 @@
+
```
Comment (rjarsenault1101) [inline lambda-functions/rds-api-cnf.json]: delete JSON

File: lambda-functions/rds-api.py [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+def make_connection():
+    return pymysql.connect(host=endpoint, user=dbuser, passwd=password,
+        port=int(port), db=database, autocommit=True, cursorclass=pymysql.cursors.DictCursor)
+
+def log_err(errmsg):
+    logger.error(errmsg)
+    return {"body": errmsg , "headers": {}, "statusCode": 400,
+        "isBase64Encoded":"false"}
+
+logger.info("Cold start complete.")
```
Comment (rjarsenault1101) [inline lambda-functions/rds-api.py]: remove

File: lambda-functions/rds-api.py [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+
+def handler(event,context):
+    #print('parameters: ', event['queryStringParameters'])
```
Comment (rjarsenault1101) [inline lambda-functions/rds-api.py]: delete

File: lambda-functions/rds-api.py [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+    TODO: sort by date
+    TODO: range selection (everything if none)
+    [DONE] TODO: time selection (everything if none)
+    [DONE] TODO: sensor selection (all sensors if none)
+    [DONE] TODO: max measurements (10 if none)
+    [DONE] TODO: measurement type (list options, all if none)
+    '''
+    parameters = event['queryStringParameters']
+    
+    if parameters is not None:
```
Comment (rjarsenault1101) [inline lambda-functions/rds-api.py:65]: Maybe add a comment describing what this cascading if is doing?

File: lambda-functions/rds-api.py [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+        '''
+        Now that we have a query statement, we need to get rid of the last AND,
+        and limit the query to the default max.
+        '''
+    else:
+        command = 'SELECT Readings.* ' + command
+    command = command[:-5] + ' ORDER BY reading_time DESC LIMIT ' + str(max_readings) + ';'
+
+    
+    print(command)
```
Comment (rjarsenault1101) [inline lambda-functions/rds-api.py]: don't need the print line

**Review by rjarsenault1101 — CHANGES_REQUESTED** [review]

**Comment by mregger** [comment]
Merging lambda functions development with the development branch. API and IoT-RDS are operational

**Pull request merged**